Content.Server/Tiles/FloorTileSystem.cs

```diff
@@ -35,10 +38,16 @@ private void OnAfterInteract(EntityUid uid, FloorTileComponent component, AfterI
 
             // this looks a bit sussy but it might be because it needs to be able to place off of grids and expand them
             var location = args.ClickLocation.AlignWithClosestGridTile();
+            foreach (var ent in location.GetEntitiesInTile(lookupSystem: _lookup))
+            {
+                // check that we the tile we're trying to access isn't blocked by a wall or something
+                if (TryComp<PhysicsComponent>(ent, out var phys) && phys.Hard && (phys.CollisionLayer & (int) CollisionGroup.Impassable) != 0)
+                    return;
```

Comment:
You should use a physicsquery + check if the body is static unless you want dynamic bodes to block it.